Develop

gaspard-rivoire opened the pull request:
Premiere pull request : 
Ajout de la page qui permet d'ajouter un devis (Il manque les 2 types de fonctionnalité d'ajout de produit)
Ajout de la page  qui permet d'ajouter un contact
Ajout de la page prospect.html qui permet de visualiser un contact (IHM d'un contact d'une entreprise mais on ne peut pas la voir parce qu'on ne peut pas encore créer un contact) 
On avance pas vite mais on essaye dur, on arrive pas a enregistre les données en no sql, on bloque sur des fonctions comme addItem.

Commit:
ajout de lu fichier prospect, modification d'affichage tier

Commit:
modification de nombreuses page, création d'une nouvelle page liste de produit pour les ajouter à une propal. Ajout d'une fonction dans app.js pour ajouter des bouton(non fonctionneles)

Commit:
ajout des boutton surla fiche des produit pour une propal

gaspard-rivoire committed:
Merge pull request #1 from FlorianMahe01/develop

ajout du fichier prospect, modification d'affichage tier

Commit:
commit 3

Commit:
essai d'ajout d'Item

gaspard-rivoire committed:
Merge branch 'develop' into develop

gaspard-rivoire committed:
Merge pull request #2 from BaptisteValayer/develop

Develop

gaspard-rivoire committed:
Merge branch 'develop' into develop

gaspard-rivoire committed:
Merge pull request #3 from FlorianMahe01/develop

essai d'ajout d'Item

atm-alexis commented on webapp/platforms/android/assets/www/app.js:
```diff
@@ -567,11 +574,43 @@ function updateItem($container, type)
 			var callback = showProposal;
 			break;
+                case 'prospect' :
+                        var callback = showProspect;
```
De ce que j'en comprends, showProspect c'est pour voir un contact... Donc showContact

aconits replied:
ça m'a l'air d'être un showThirdparty qui est 6 lignes plus haut

atm-alexis commented on webapp/platforms/android/assets/www/indexdb.js:
```diff
@@ -271,6 +272,50 @@ var DoliDb = function() {
 	};
+        
+        
+        DoliDb.prototype.createProspect = function(id_object, fk_soc) {
```
idem createContact

aconits replied:
createItem existant

atm-alexis commented on webapp/platforms/android/assets/www/indexdb.js:
```diff
@@ -271,6 +272,50 @@ var DoliDb = function() {
+        DoliDb.prototype.createProspect = function(id_object, fk_soc) {
+		
+		if (typeof fk_soc == 'undefined' || !fk_soc) { 
+			showMessage('Warning', 'Can\'t create a proposal without thirdparty id', 'warning');
```
Can't create **contact** without thirdparty_id...

atm-alexis commented on webapp/platforms/android/assets/www/indexdb.js:
```diff
@@ -271,6 +272,50 @@ var DoliDb = function() {
+		}
+		
+		var obj = {
+			ref: '(PROS'+($.now())+')'
```
(PROV -S +V

atm-alexis commented on webapp/platforms/android/assets/www/indexdb.js:
```diff
@@ -271,6 +272,50 @@ var DoliDb = function() {
+			,update_by_indexedDB:0
+		};
+                
+  console.log('buozad',obj);
```
attention aux traces

atm-alexis commented on webapp/www/app.js:
```diff
@@ -567,23 +576,68 @@ function updateItem($container, type)
+    else 
+       param.val(0);
+}
+function testconsole(){
```
ouais pourquoi pas

atm-alexis commented on webapp/www/app.js:
```diff
@@ -567,23 +576,68 @@ function updateItem($container, type)
+function addbouton(){
+    $('ul.list-group.list_product').children('.list-group-item').append('<span class="AddListBtn" style="float:right;"><button class="btn-circle btn btn-warning" type="button" onclick="modifySelect(this)" value=0><span class="glyphicon glyphicon-plus-sign" aria-hidden="true"></span></button></span>');    
+}
 
```
ça manque de commentaire, que font ces fonctions et surtout pour quelles raisons

atm-alexis commented on webapp/www/indexdb.js:
```diff
@@ -271,6 +270,46 @@ var DoliDb = function() {
 		
 	};
+        
+        DoliDb.prototype.createProspect = function(id_object, fk_soc) {
```
Bon alors je viens de voir que vous aviez commité le projet cordova... git add * ce n'est pas conseillé, surtout pour une première review, ça me simplifie pas la lecture

atm-alexis commented on webapp/www/tpl/newfile:
```diff
@@ -0,0 +1,13 @@
+el=document.getElementById('product-list-propal').firstElementChild.firstElementChild.nextElementSibling.firstElementChild/*.firstElementChild*/;
```
il sert à qq chose ce fichier ?

atm-alexis commented on webapp/www/tpl/propal_product.html:
```diff
@@ -0,0 +1,23 @@
+<body  onload="addbouton()"><div id="product-list-propal" class="tab-pane">
```
Si le addbouton (addButton !) est ajouté au onload... Pourquoi donc ajouter ce bouton en JS et ne pas simplement l'ajouter en HTML ?

atm-alexis commented on webapp/www/tpl/proposal.html:
```diff
@@ -53,6 +53,146 @@ <h1 rel="ref"></h1>
+				<label for="DateValidite"><langs trans="DateDeValidite"></langs></label>
+				<div id="DateValidite">
+					<select class="form-control" name="duree_validite">
+                                                <option value="" disabled selected>Choisir la duree de validite</option>
```
C'est option doivent venir de Dolibarr et stocker en BD locale à la synchro.

atm-alexis commented on webapp/www/tpl/proposal.html:
```diff
@@ -53,6 +53,146 @@ <h1 rel="ref"></h1>
+				<label for="ModeReglement"><langs trans="ProposalReglement"></langs></label>
+				<div id="ModeReglement">
+					<select class="form-control" name="mode_reglement">
+						<option value="" disabled selected>Choisir le mode de paiement</option>
```
idem

atm-alexis commented on webapp/www/tpl/proposal.html:
```diff
@@ -173,7 +313,7 @@ <h1 rel="ref"></h1>
 	<div class="row navbar-fixed-bottom">
 		<a data-toggle="tab" href="#proposal-card" class="btn btn-default col-xs-6 move_tab"><langs trans="Cancel"></langs></a>
-		<a data-toggle="tab" href="#proposal-card" onclick="updateItem($('#proposal-card-edit'), 'proposal');" class="btn btn-success col-xs-6 move_tab"><langs trans="Update"></langs></a>
+		<a data-toggle="tab" href="#proposal-card" onclick="updateItem($('#proposal-card-add'), 'proposal');" class="btn btn-success col-xs-6 move_tab"><langs trans="Add proposal"></langs></a>
```
pas d'espace dans les clefs de traduction

atm-alexis commented on webapp/www/tpl/prospect.html:
```diff
@@ -0,0 +1,174 @@
+				</div>
+				
+				<!-- Phone-->
+				<label for="phone" class="col-sm-2 control-label"><langs trans="Phone"></langs></label>
```
un contact à 3 numéros de tel, phone_perso, phone_pro et phone_portable de mémoire. Voir pour une manière simple d'afficher les 3

atm-alexis commented on webapp/www/tpl/thirdparty.html:
```diff
@@ -88,44 +105,43 @@ <h1 rel="name"></h1>
 				
 				<!-- Prospect/Client -->
-				<label for="prospect" class="col-sm-2 control-label"><langs trans="ThirdpartyProspect"></langs></label>
+				<label for="prospect" class="col-sm-2 control-label"><langs trans="Prospects"></langs></label>
```
ça indique le type de tiers

atm-alexis commented on webapp/www/tpl/thirdparty.html:
```diff
@@ -135,30 +151,30 @@ <h1 rel="name"></h1>
 						</option>
 						<option>
-							oui
+							Yes
```
Attention : <langs trans="Yes"></langs>

Commit:
commit 2

gaspard-rivoire committed:
Merge pull request #4 from FlorianMahe01/develop

commit 2

Commit:
fonction addbouton fonctionnel dans la liste des products pour le propal

gaspard-rivoire committed:
Merge pull request #5 from BaptisteValayer/develop

fonction addbouton fonctionnel dans la liste des products pour le propal

aconits commented on webapp/platforms/android/assets/www/app.js:
```diff
@@ -567,11 +574,43 @@ function updateItem($container, type)
 	doliDb.updateItem(type, id, TValue, callback);
 }
 
+function addItem($container, type)
```
hmmm ... cf createItem et updateItem

atm-ph commented:
Attention vous êtes en train de modifier 
 - webapp/platforms/android/assets/www/app.js
 - webapp/www/app.js

Je sais pas si vous recompilez avec cordova à la volée pour tester sur téléphone (plutôt que via navigateur) mais si c'est pas le cas, faut modifier à partir de www/

BaptisteValayer committed:
Merge pull request #1 from gaspard-rivoire/develop

Develop

Commit:
add a test, remove addItem, create createItem

Commit:
Split de la classe app.js/Avancement de l'IHM propal

app.js contient les fonctions général de l'application
appOtherDocument contient les fonctions specifique au document
(Bill/Order/Contact) on pourrait dissocier Contact dans un class
appContact par exemple
appProduct.js contient les fonctions specifique aux produits
appPropal.js contient les fonctions specifique aux Propals
appThirdParty.js contient les fonctions specifique aux ThirdPart
Debut de documentation des fonctions de app.js

gaspard-rivoire committed:
Merge pull request #6 from BaptisteValayer/develop

Develop

FlorianMahe01 committed:
Merge branch 'develop' into develop

gaspard-rivoire committed:
Merge pull request #7 from FlorianMahe01/develop

Pull Request 7

aconits commented on webapp/www/test/test.html:
```diff
@@ -0,0 +1,38 @@
+  ,desc:"Bizuland"  
+};
+
+DB.createItem('product', item, showAll);
```
dans webapp/www/index.html il y a une instanciation de l'objet 
 - doliDb = new DoliDb();

Du coup c'est la variable doliDb qu'il faut utiliser

atm-alexis replied:
pff PH aprend à lire ;)

Commit:
save version, modifi createItem, we can now create a contact

Commit:
tantative d'issertion de contact dans la bdd:

gaspard-rivoire committed:
Merge pull request #8 from FlorianMahe01/develop

Develop

Commit:
commit 8

gaspard-rivoire committed:
Merge branch 'develop' into develop

gaspard-rivoire committed:
Merge pull request #9 from BaptisteValayer/develop

commit 8
